cocotb/log.py

```diff
@@ -192,6 +192,9 @@ class SimColourLogFormatter(SimLogFormatter):
     def format(self, record):
         """Prettify the log output, annotate with simulation time"""
 
+        # To be able to format them, messages must be strings.
+        record.msg = str(record.msg)
+
         if record.args:
             msg = record.msg % record.args
         else:
```

Comment:
Correct fix I think is to replace all these lines with `msg = record.getMessage()`, as is done [by python in the default logger](https://github.com/python/cpython/blob/4c1b6a6f4fc46add0097efb3026cf3f0c89f88a2/Lib/logging/__init__.py#L651-L664).

Reply:
Same fix is needed on line 173

Reply:
Good find, updated.